Replicator/Pusher+Revs.cc

```diff
@@ -76,11 +83,25 @@ namespace litecore::repl {
                                                       _options.propertyEncryptor,
                                                       _options.callbackContext,
                                                       &c4err);
-            if (encryptedRoot)
+            if (encryptedRoot) {
                 root = encryptedRoot;
-            else if (c4err) {
+            } else if (c4err.domain == WebSocketDomain && c4err.code == 503) {
+                // This is treated as a transient network glitch, we lift it to the replicator
+                // to handle. The replicator will be taken to offline and restarted after a certain
+                // wait time.
+                onError(c4err);
```

Comment:
To be consistent with the pull replication that also reports the error to the document replication event, `finishedDocumentWithError(request, c4err, false)` should be called as well. 

Noted that before making this change, the previous change also calls `finishedDocumentWithError(request, c4err, false)` when getting the 503 error.

Reply:
Oh, my bad. Refactoring mistake.

Reply:
Maybe the test should include the callback logic?

Reply:
Good idea